consumer/src/mode/decoded/deposited_curve.rs

```diff
@@ -0,0 +1,333 @@
+                triple_id: curve_vault.triple_id.clone(),
+                curve_number: curve_vault.curve_number.clone(),
+                total_shares: curve_vault.total_shares.clone(),
+                current_share_price: curve_vault.current_share_price.clone(),
+                position_count: curve_vault.position_count + 1,
+            };
+            
+            // Update the curve vault
+            updated_curve_vault
+                .upsert(
+                    &decoded_consumer_context.pg_pool,
+                    &decoded_consumer_context.backend_schema,
+                )
+                .await
+                .map_err(|e| ConsumerError::ModelError(e))?;
+        }
+
+        Ok(())
+    }
+
+    /// This function creates a `Signal` for the `DepositedCurve` event
+    async fn create_signal(
+        &self,
+        decoded_consumer_context: &DecodedConsumerContext,
+        event: &DecodedMessage,
+        curve_vault: &CurveVault,
+    ) -> Result<(), ConsumerError> {
+        if self.senderAssetsAfterTotalFees > U256::from(0) {
+            if let Some(atom_id) = curve_vault.atom_id.clone() {
+                Signal::builder()
+                    .id(DecodedMessage::event_id(event))
+                    .account_id(self.sender.to_string().to_lowercase())
+                    .delta(U256Wrapper::from(self.senderAssetsAfterTotalFees))
+                    .atom_id(atom_id)
+                    .deposit_id(DecodedMessage::event_id(event))
+                    .block_number(U256Wrapper::try_from(event.block_number)?)
+                    .block_timestamp(event.block_timestamp)
+                    .transaction_hash(event.transaction_hash.clone())
+                    .build()
+                    .upsert(
+                        &decoded_consumer_context.pg_pool,
+                        &decoded_consumer_context.backend_schema,
+                    )
+                    .await?;
+            } else if let Some(triple_id) = curve_vault.triple_id.clone() {
+                Signal::builder()
+                    .id(DecodedMessage::event_id(event))
+                    .account_id(self.sender.to_string().to_lowercase())
+                    .delta(U256Wrapper::from(self.senderAssetsAfterTotalFees))
+                    .triple_id(triple_id)
+                    .deposit_id(DecodedMessage::event_id(event))
+                    .block_number(U256Wrapper::try_from(event.block_number)?)
+                    .block_timestamp(event.block_timestamp)
+                    .transaction_hash(event.transaction_hash.clone())
+                    .build()
+                    .upsert(
+                        &decoded_consumer_context.pg_pool,
+                        &decoded_consumer_context.backend_schema,
+                    )
+                    .await?;
+            }
+        }
+        Ok(())
+    }
+
+    /// This function creates a `Deposit` record for the `DepositedCurve` event
+    async fn create_deposit(
```

Comment:
For curve deposits and redeems, we want to keep track of a "total" ledger for a given atom, triple, counter-triple.  So that way we can query based on the TVL of the atom itself, rather than being stuck with the TVL of individual vaults.

This is one of the things which is not yet done in this branch -- need to add it.

Reply:
This would look like having a vaults_totals item that updates the total shares and total assets whenever any constituent vault balance changes (i.e. curve 2 balance went up, so we increment the vaults_totals by that amount, etc)